src/Microsoft.SqlTools.ServiceLayer/Scripting/ScripterCore.cs

```diff
@@ -453,7 +446,71 @@ internal IEnumerable<Declaration> GetCompletionsForToken(ParseResult parseResult
             return Resolver.FindCompletions(
                 parseResult, parserLine, parserColumn, metadataDisplayInfoProvider);
         }
-        #endregion
 
+        /// <summary>
+        /// Wrapper method that calls Resolver.FindCompletions
+        /// </summary>
+        /// <param name="objectName"></param>
+        /// <param name="schemaName"></param>
+        /// <param name="objectType"></param>
+        /// <param name="tempFileName"></param>
+        /// <returns></returns>
+        internal ScriptingScriptOperation InitScriptOperation(string objectName, string schemaName, string objectType, string tempFileName)
+        {            
+            // object that has to be scripted
+            ScriptingObject obj = new ScriptingObject {
+                Name = objectName,
+                Schema = schemaName,
+                Type = objectType
+            };
+
+            // scripting options
+            ScriptOptions options = new ScriptOptions {
+			    ScriptCreateDrop = "ScriptCreate",
+			    TypeOfDataToScript = "SchemaOnly",
+			    ScriptStatistics = "ScriptStatsNone",
+			    TargetDatabaseEngineEdition = GetTargetDatabaseEngineEdition(),
+			    TargetDatabaseEngineType = GetTargetDatabaseEngineType(),
+			    ScriptCompatibilityOption = GetScriptCompatibilityOption(),
+                IncludeIfNotExists = true
+		    };
+
+            List<ScriptingObject> objList = new List<ScriptingObject>();
+            List<string> inclList = new List<string>();
+            inclList.Add(objectType);
+            objList.Add(obj);
+
+            // create parameters for the scripting operation
+
+            ScriptingParams parameters = new ScriptingParams {
+                FilePath = tempFileName,
+                ConnectionString = ConnectionService.BuildConnectionString(this.connectionInfo.ConnectionDetails),
+                IncludeTypes = inclList,
+                ScriptingObjects = objList,
+                ScriptOptions = options
+            };
+
+            return new ScriptingScriptOperation(parameters);
+        }
+
+        internal string GetTargetDatabaseEngineEdition() 
+        {
+            DatabaseEngineEdition dbEngineEdition = this.serverConnection.DatabaseEngineEdition;
+            string dbEngineEditionString = targetDatabaseEngineEditionMap[dbEngineEdition];
+            return (dbEngineEditionString != null) ? dbEngineEditionString : "SqlServerEnterpriseEdition";
+        }
+
+        internal string GetScriptCompatibilityOption()
+        {
+            int serverVersion = this.serverConnection.ServerVersion.Major;
+            string dbEngineTypeString = serverVersionMap[serverVersion];
```

Comment:
>serverVersionMap[serverVersion] [](start = 40, length = 31)

any need to check the map contains this value prior to indexing, or we know the map will always contain all possible values?

Reply:
Map contains all the possible values.